Classes/Code/ScriptTagBuilder.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Code;
+
+use Brotkrueml\MatomoIntegration\Event\EnrichScriptTagEvent;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * The tag builder creates the header script tag to include matomo
+ *
+ * @internal
+ */
+class ScriptTagBuilder
+{
+    private ServerRequestInterface $request;
+    private EventDispatcherInterface $eventDispatcher;
+
+    public function __construct(
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * Build the script start script tag with attributes
```

Comment:
"start script" is superfluous here.
Perhaps: Build the script tag with optional attributes.

Reply:
Updated